cli/towerSysUtil.ml

```diff
@@ -103,5 +105,41 @@ module Dir = struct
       match err_opt with None -> make_exist ~perm path | Some err -> Some err
     ) None
 
+  (* Placeholder for now *)
+  let log msg = ()
+
+  let gather_nodes root =
+    (* By default enforce all recursive processing is done relative to
+    the cwd dir, unless overridden *)
+    let cwd = Unix.getcwd() in
+    let rec gather_nodes node =
+      let node_path = cwd ^ "/" ^ node in
+      match (Unix.lstat node_path).st_kind with
+      | Unix.S_REG -> [|node|]
+      | Unix.S_DIR -> let subnodes = (Sys.readdir node) in
+                      let descend subnode =
+                        let subnode_path = node ^ "/" ^ subnode in
+                        gather_nodes subnode_path
+                      in
+                      Array.fold_left Array.append [|node|] (Array.map descend subnodes)
+      | _ -> [||]
+    in
+    gather_nodes root
+
+  let rmdir_r path =
+    if Sys.file_exists path then
+      let remove_node node =
+        if Sys.file_exists node then
+          match (Unix.lstat node).st_kind with
+          | Unix.S_DIR -> Unix.rmdir node
+          | Unix.S_REG -> Unix.unlink node
+          | _ -> Unix.unlink node
+      in
+      (* Walk the full tree first, then process results, prevent
+      partially-computed results in case of an error walking. *)
+      let unsorted_nodes = Array.to_list (gather_nodes path) in
+      let nodes = List.sort (fun a b -> (String.length b) - (String.length a)) unsorted_nodes in
```

Comment:
I don't think this sort is necessary if the subtree/leaf nodes always precede the parent node (by consing the parent onto the accumulator before folding/passing it to recursive calls.